chains/solana/contracts/programs/execution-buffer/src/context.rs

```diff
@@ -0,0 +1,113 @@
+use crate::state::BufferedReport;
+use anchor_lang::prelude::*;
+use ccip_common::seed;
+
+pub const ANCHOR_DISCRIMINATOR: usize = 8; // size in bytes
+
+#[derive(Accounts)]
+#[instruction(buffer_id: u64, data: Vec<u8>)]
+pub struct AppendExecutionReportData<'info> {
+    #[account(
+        mut,
+        seeds = [seed::EXECUTION_BUFFER, authority.key().as_ref(), &buffer_id.to_le_bytes()],
```

Comment:
I'd still prefer to use a 32-byte value here instead of u64. That way, the caller may choose to use the msgId as a key for the buffer if they want to. By not naming it explicitly `msg_id`, you still leave it open to reuse in other cases, if you want this implementation to be usable later for commits as well (where the caller could use either the root or something like chainSelector&seqNum)